plugins/examples/custom-processor/src/main/java/org/elasticsearch/example/customprocessor/ExampleRepeatProcessor.java

```diff
@@ -0,0 +1,48 @@
+package org.elasticsearch.example.customprocessor;
+
+import org.elasticsearch.ingest.AbstractProcessor;
+import org.elasticsearch.ingest.IngestDocument;
+import org.elasticsearch.ingest.Processor;
+
+import java.util.Map;
+
+/**
+ * Example of adding an ingest processor with a plugin.
+ */
+public class ExampleRepeatProcessor extends AbstractProcessor {
+    public static final String TYPE = "repeat";
+    public static final String FILED_TO_REPEAT = "to_repeat";
+
+    ExampleRepeatProcessor(String tag, String description) {
+        super(tag, description);
+    }
+
+    @Override
+    public IngestDocument execute(IngestDocument document) {
+        Object val = document.getFieldValue(FILED_TO_REPEAT, Object.class, true);
+
+        if (val instanceof String string) {
+            String repeated = string.concat(string);
+            document.setFieldValue(FILED_TO_REPEAT, repeated);
+        }
+        return document;
+    }
+
+    @Override
+    public String getType() {
+        return TYPE;
+    }
+
+    public static class Factory implements Processor.Factory {
+
+        @Override
+        public ExampleRepeatProcessor create(
+            Map<String, Processor.Factory> registry,
+            String tag,
+            String description,
+            Map<String, Object> config
+        ) {
+            return new ExampleRepeatProcessor(tag, description);
```

Comment:
It might be worth demonstrating how a user can pass properties into the processor here. For instance, most ES processors would have you specify which field name on a document to repeat. Maybe making a change so that the processor definition accepts a configuration parameter called `field` in the definition that will later be used to identify which document field name to repeat instead of having a single property name that is expected by the processor.

Reply:
I thought about demonstrating passing in field name, and supporting Mustache template for the field name as well. But then I thought maybe it's better to keep this to minimum required for a processor plugin to avoid confusion. 

I am thinking we keep this example simple, and add a reference link to an existing processor (like [Set](https://github.com/elastic/elasticsearch/blob/main/modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/SetProcessor.java)) that demonstrates passing in the field name and other features like Mustache.

What do you think? I am open to either way.

Reply:
We probably don't need to involve using mustache in the example plugin, but using some values from the config object to show how they might be organized would be helpful.